### PR TITLE
Add type hints to test.tz_stub

### PR DESCRIPTION
**Pull request by hramezani**

### CODE REVIEW

**Review by sethmlarson — DISMISSED** [review]
LGTM

**Comment by sethmlarson** [comment]
GitHub Actions are broken right now: https://www.githubstatus.com/

**Comment by hramezani** [comment]
> GitHub Actions are broken right now: https://www.githubstatus.com/

Thanks for letting me know. I will try to run them again tomorrow!

**Comment by hramezani** [comment]
@sethmlarson `mypy --install-types --non-interactive` failed with this error:
`error: Can't determine which types to install with no files to check (and no cache from previous mypy run)`
So, I've installed `types-python-dateutil` based on `mypy` suggestion.
```
test/tz_stub.py:8: note: Hint: "python3 -m pip install types-python-dateutil"
test/tz_stub.py:8: note: (or run "mypy --install-types" to install all missing stub packages)
```
Please take another look.
Thanks!

File: test/tz_stub.py [hunk 1]
```diff
@@ -36,5 +37,5 @@ def stub_timezone_ctx(tzname):
     os.environ["TZ"] = tzname
     time.tzset()
     yield
-    os.environ["TZ"] = old_tzname
+    os.environ["TZ"] = old_tzname  # type: ignore[assignment]
```
Comment (bluetech) [inline test/tz_stub.py:40]: What is the type of `old_tzname` here?

**Comment by bluetech** [comment]
I wonder if we can use the standard `zoneinfo` module instead of `dateutil`. The dateutil usage here seems kinda optional, like some sanity test IIUC. So we can either just remove it for py<3.9 or use `backports.zoneinfo` for older versions.

File: test/tz_stub.py (hunk 1 above)
Comment (hramezani) [reply test/tz_stub.py:40]: here is the error message from mypy:
`Incompatible types in assignment (expression has type "Optional[str]", target has type "str")  [assignment]`

The `old_tzname` comes from `old_tzname = datetime.datetime.now(local_tz).tzname()`.
`tzname()` return type is `Optional[str]`

Comment (bluetech) [reply test/tz_stub.py:40]: Ha got it. I just checked, if we assign `None` to `os.environ` it raises a `TypeError`. So I think what we should do is `assert old_tzname is not None` right after obtaining it, to clarify our assumption and avoid the type-ignore.

Comment (hramezani) [reply test/tz_stub.py]: Good suggestion. Thanks!
I've updated the patch based on your suggestion.

**Review by bluetech — APPROVED** [review]

**Pull request merged**